src/devices/SDLJoypad/SDLJoypad.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2016 iCub Facility, Istituto Italiano di Tecnologia
+ * Authors: Andrea Ruzzenenti <andrea.ruzzenenti@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include "SDLJoypad.h"
+#include <stdio.h>
+#include <yarp/os/LogStream.h>
+
+
+using namespace yarp::dev;
+using namespace std;
+SDLJoypad::SDLJoypad()
+{
+    m_buttonCount = 0;
+    m_axisCount   = 0;
+    m_stickCount  = 0;
+    m_touchCount  = 0;
+    m_ballCount   = 0;
+    m_hatCount    = 0;
+    m_device      = 0;
+}
+
+SDLJoypad::~SDLJoypad()
+{
+
+}
+
+bool SDLJoypad::open(yarp::os::Searchable& rf)
+{
+    int joy_id, joystick_num;
+    if (SDL_InitSubSystem( SDL_INIT_JOYSTICK ) < 0 )
+    {
+        yError ( "SDLJoypad: Unable to initialize Joystick: %s\n", SDL_GetError() );
+        return false;
+    }
+
+    joy_id       = 0;
+    joystick_num = SDL_NumJoysticks();
+    if (joystick_num == 0)
+    {
+        yError ( "SDLJoypad: No joysticks found\n");
+        return false;
+    }
+    else if (joystick_num == 1)
+    {
+        joy_id = 0;
+        yInfo ( "SDLJoypad: One joystick found \n");
+        yInfo ( "SDLJoypad: Using joystick: %s \n", SDL_JoystickName(joy_id));
+    }
+    else
+    {
+        yInfo ( "SDLJoypad: More than one joystick found:\n");
+        for (int i = 0; i < joystick_num; i++)
+        {
+            yInfo ( "%d: %s\n", i, SDL_JoystickName(i));
+        }
+        yInfo ( "\n");
```

Comment:
yes it is. i will delete it (it was like this in the JoystickCtrl of icub but i don't know why)